daal4py/sklearn/neighbors/knn_classifier.py

```diff
@@ -0,0 +1,546 @@
+        kneighbors_graph = csr_matrix((A_data, A_ind.ravel(), A_indptr),
+                                      shape=(n_queries, n_samples_fit))
+
+        return kneighbors_graph
+
+
+class KNeighborsClassifier(NeighborsBase, KNeighborsMixin,
+                           SupervisedIntegerMixin, ClassifierMixin):
+
+    @_deprecate_positional_args
+    def __init__(self, n_neighbors=5, *,
+                 weights='uniform', algorithm='auto', leaf_size=30,
+                 p=2, metric='minkowski', metric_params=None, n_jobs=None,
+                 **kwargs):
+        super().__init__(
+            n_neighbors=n_neighbors,
+            algorithm=algorithm,
+            leaf_size=leaf_size, metric=metric, p=p,
+            metric_params=metric_params,
+            n_jobs=n_jobs, **kwargs)
+        self.weights = _check_weights(weights)
+
+    def predict(self, X):
+        """Predict the class labels for the provided data.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_queries, n_features), \
+                or (n_queries, n_indexed) if metric == 'precomputed'
+            Test samples.
+
+        Returns
+        -------
+        y : ndarray of shape (n_queries,) or (n_queries, n_outputs)
+            Class labels for each data sample.
+        """
+        X = check_array(X, accept_sparse='csr')
+
+        try:
+            fptype = getFPType(X)
+        except ValueError:
+            fptype = None
+
+        if self.weights in ['uniform', 'distance'] and self.algorithm in ['brute'] \
+        and (self.metric == 'minkowski' and self.p == 2 or self.metric == 'euclidean') \
+        and self._y.ndim == 1 and fptype is not None:
+
+            n_classes = len(self.classes_)
+
+            if self.algorithm == 'brute':
+                knn_classification_training = d4p.bf_knn_classification_training
+                knn_classification_prediction = d4p.bf_knn_classification_prediction
+            else:
+                knn_classification_training = d4p.kdtree_knn_classification_training
+                knn_classification_prediction = d4p.kdtree_knn_classification_prediction
+
+            training_alg = knn_classification_training(
+                fptype=fptype,
+                method='defaultDense',
+                k=self.n_neighbors,
+                nClasses=n_classes
+            )
+
+            fit_X = d4p.get_data(self._fit_X)
+            training_result = training_alg.compute(fit_X, self._y.reshape(self._y.shape[0], 1))
+
+            prediction_alg = knn_classification_prediction(
+                fptype=fptype,
+                method='defaultDense',
+                k=self.n_neighbors,
+                nClasses=n_classes
+            )
+
+            X = d4p.get_data(X)
+            prediction_result = prediction_alg.compute(X, training_result.model)
+            return prediction_result.prediction.ravel().astype(self._y.dtype)
+
+        else:
+            neigh_dist, neigh_ind = self.kneighbors(X)
```

Comment:
Why are you copying everything? See how it is done in svm. The base class fit is simply called there if we cannot patch.

Reply:
Changed